app/src/ui/app.tsx

```diff
@@ -1190,6 +1216,61 @@ export class App extends React.Component<IAppProps, IAppState> {
     })
   }
 
+  private openPullRequest() {
+    const selection = this.state.selectedState
+
+    if (!selection || selection.type !== SelectionType.Repository) {
+      return
+    }
+
+    const tip = selection.state.branchesState.tip
+
+    if (tip.kind !== TipState.Valid) {
+      return
+    }
+
+    const dispatcher = this.props.dispatcher
+    const repository = selection.repository
+    const branch = tip.branch
+    const aheadBehind = selection.state.aheadBehind
+
+    if (!aheadBehind) {
+      //Pubilish branch
+      dispatcher.showPopup({
+        type: PopupType.PublishBranch,
+        repository,
+        branch,
+      })
+    } else if (aheadBehind.ahead > 0) {
+      //Push commits
+      dispatcher.showPopup({
+        type: PopupType.PushBranchCommits,
+        repository,
+        branch,
+        unPushedCommits: aheadBehind.ahead,
+      })
+    } else {
+      this.OpenPullRequestOnGithub(repository, branch)
+    }
+  }
+
+  private OpenPullRequestOnGithub = (
+    repository: Repository,
+    branch: Branch
+  ) => {
+    const gitHubRepository = repository.gitHubRepository
+
+    if (!gitHubRepository) {
+      return
+    }
+
+    const baseURL = `${gitHubRepository.htmlURL}/compare/${gitHubRepository.defaultBranch}...${branch.nameWithoutRemote}`
+
+    if (baseURL) {
```

Comment:
At the time it was, now I need to make sure ``${gitHubRepository.htmlURL}` is not null or empty or undefined or 0 or w/e JS sees as false.